### PR TITLE
ci: Make version bumping script more robust

### PR DESCRIPTION
**Pull request by achals**
Signed-off-by: Achal Shah <achals@gmail.com>

Running locally is smooth:

```
$ python ./infra/scripts/release/bump_file_versions.py 0.21.0 0.22.0
Updated 12 files with new version 0.22.0
[13/05/22 3:10:09] (python-3.8.12) ~/tecton/feast   achal/update-version-bump-script ● 
$ git st
On branch achal/update-version-bump-script
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   infra/charts/feast-python-server/Chart.yaml
	modified:   infra/charts/feast-python-server/README.md
	modified:   infra/charts/feast/Chart.yaml
	modified:   infra/charts/feast/README.md
	modified:   infra/charts/feast/charts/feature-server/Chart.yaml
	modified:   infra/charts/feast/charts/feature-server/README.md
	modified:   infra/charts/feast/charts/feature-server/values.yaml
	modified:   infra/charts/feast/charts/transformation-service/Chart.yaml
	modified:   infra/charts/feast/charts/transformation-service/README.md
	modified:   infra/charts/feast/charts/transformation-service/values.yaml
	modified:   infra/charts/feast/requirements.yaml
	modified:   infra/scripts/release/bump_file_versions.py
	modified:   java/pom.xml
```

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: infra/scripts/release/bump_file_versions.py [hunk 1]
```diff
@@ -33,46 +33,22 @@ def main() -> None:
...
-                current_version = current_version_patch
-                found = True
-                break
-            else:
-                print(f"Found {versions_in_files} occurrences of {current_version_patch}, instead of {VERSIONS_TO_BUMP}")
-        if not found:
-            raise SystemExit(f"Could not find {VERSIONS_TO_BUMP} versions of {current_version} in {path_to_file_list}")
-
-    print(f"Found {versions_in_files} occurrences of {current_version} in files to bump {path_to_file_list}")
+    # The current version should be 0.18.0 or 0.19.0 or 0.20.0 etc
```
Comment (kevjumba) [inline infra/scripts/release/bump_file_versions.py:36]: Don't think we need 0.18.0 or 0.19.0 right? Also after version bump we're at 0.21?

File: infra/scripts/release/bump_file_versions.py [hunk 2]
```diff
@@ -88,14 +64,18 @@ def is_semantic_version(version: str) -> bool:
...
-            total += file_contents.count(current_version)
-    return total
+        components = file.split(" ")
+        assert len(components) > 1, f"Entry {file} should have a file name, and a list of line numbers with versions"
+        file_path = components[0]
+        lines = components[1:]
+        with open(repo_root.joinpath(file_path), "r") as f:
+            file_contents = f.readlines()
+            for line in lines:
+                assert current_version in file_contents[int(line) - 1], f"File {file_path} line {line} didn't containe " \
```
Comment (kevjumba) [inline infra/scripts/release/bump_file_versions.py:76]: nit: "contain"

File: infra/scripts/release/bump_file_versions.py (hunk 1 above)
Comment (achals) [reply infra/scripts/release/bump_file_versions.py:36]: Just as an example, but I'll update.

File: infra/scripts/release/bump_file_versions.py (hunk 2 above)
Comment (achals) [reply infra/scripts/release/bump_file_versions.py:76]: oops fixed

**Review by adchia — APPROVED** [review]
/lgtm

**Pull request merged**